src/Microsoft.AspNet.OData.Shared/Builder/InstanceAnnotationContainerAnnotation.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Reflection;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    /// <summary>
+    /// This annotation indicates the mapping from a <see cref="IEdmStructuredType"/> to a <see cref="PropertyInfo"/>.
+    /// The <see cref="IEdmStructuredType"/> is a type of IODataInstanceAnnotationContainer and the <see cref="PropertyInfo"/> is the specific
+    /// property which is used to save/retrieve the instance annotations.
+    /// </summary>
+    public class ODataInstanceAnnotationContainerAnnotation
```

Comment:
>public [](start = 4, length = 6)

Does this class have to be public, or can it be internal? The developer never sees it, right?  Does it have to be public because of how we add it to the model?